webapp/src/snippetBuilder.tsx

```diff
@@ -0,0 +1,398 @@
+/// <reference path="../../built/pxtlib.d.ts" />
+
+import * as React from "react";
+import * as data from "./data";
+import * as sui from "./sui";
+import * as md from "./marked";
+import * as compiler from './compiler';
+
+type ISettingsProps = pxt.editor.ISettingsProps;
+
+export interface SnippetBuilderState {
+    visible?: boolean;
+    tsOutput?: string;
+    projectView?: pxt.editor.IProjectView;
+    answers?: any; // Will be typed once more clearly defined
```

Comment:
avoid any even in early phases of the projects. it defeats the benefits of typescript. define interfaces and evolve them as you build the solution.

Reply:
Updated most of the types to use descriptive interfaces.